pkg/dns/config/sync.go

```diff
@@ -47,50 +36,28 @@ type Sync interface {
 	Periodic() <-chan *Config
 }
 
+type SyncResult struct {
+	Version string
+	Data    map[string]string
+}
+
+type SyncSource interface {
```

Comment:
I think SyncResult for sure does not need to be exposed. I just tried it with your patch and there isn't an issue.

It would be preferable to just have just expose NewConfigMapSync and NewFileSync constructors for SYnc and hide the SyncSource and SyncResult from outside altogether.